hypex/comparators/abstract.py

```diff
@@ -47,51 +65,46 @@ def _inner_function(
         raise AbstractMethodError
 
     def _get_fields(self, data: ExperimentData):
-        group_field = self._field_searching(data, self.grouping_role)
+        tmp_role = True if data.ds.tmp_roles else False
+        group_field = self._field_searching(
+            data=data,
+            field=self.grouping_role,
+            search_types=self.search_types,
+            space=self.space,
+        )
         target_fields = self._field_searching(
             data,
             TempTargetRole(),
```

Comment:
Why TmpTarget? If we talk about OnRoleExperiment usage, it is ok, but in other situations target_filds won't be found.